dll/win32/shell32/folders/CFSFolder.cpp

```diff
@@ -1082,7 +1089,9 @@ HRESULT WINAPI CFSFolder::CompareIDs(LPARAM lParam,
     /* When sorting between a File and a Folder, the Folder gets sorted first */
     if (bIsFolder1 != bIsFolder2)
     {
-        return MAKE_COMPARE_HRESULT(bIsFolder1 ? -1 : 1);
+        // ...but only if neither of them were generated by SHSimpleIDListFromPath
```

Comment:
I wonder what the underlying reason for that choice is.

Reply:
The reason to sort folders like this or my change?

For my change, PIDLs created by `SHSimpleIDListFromPath` (used by `SHChangeNotify` when given a path) does not contain any information other than the name. This is problematic when you compare a real item (with full information) vs a simple item like this. This will not affect sorting in Explorer since both of them are real in that case.

This is an unfortunate design decision from Windows 95. `SHChangeNotify` does technically have information about the item being a file or a folder (DELETE vs RMDIR parameter) even after it has been deleted but there are probably other cases where no information about the item exists because it's too early (not created yet) or too late (already deleted).

Reply:
The reason of the "but only if neither of them were generated by SHSimpleIDListFromPath", yes.